code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardUIProvider.kt

```diff
@@ -121,7 +143,14 @@ class ContentCardUIProvider(val surface: Surface) : AepUIContentProvider {
                             Log.error(
                                 MessagingConstants.LOG_TAG,
                                 SELF_TAG,
-                                "Failed to build template: ${e.message}"
+                                "Failed to build template: proposition ID : ${proposition.uniqueId} ${e.message}"
+                            )
+                            completion(
+                                Result.failure(
```

Comment:
Good point. We shouldn't be calling completion block multiple times on failure for a single call to `getAepUITemplateList`.  We should align on one of the following:
- continue the loop and parse remaining propositions instead of calling completion when an individual proposition parsing fails in the list. Then, return with `Result.success` if at least one has been successful. Else return `Result.failure`.
- Return `Result.success` if all of the proposition parsing succeeds.

I am inclined to do the first option, because the consumer cares only if nothing is displayed.